File: sol/2405.cpp
struct Solution {
    inline int partitionString(string &s) {
        int res = 1, h = 0;
        for (char c : s){
            int m = 1 << (c - 'a');
            if (h & m){
                ++res;
                h = 0;
            }
            h |= m;
        }
        return res;
    }
};

File: sol/2405/2405.py
class Solution:
    def partitionString(self, s: str) -> int:
        res = 1
        counter = 0
        for c in s:
            m = 1 << (ord(c) - ord('a'))
            if (counter & m) != 0:
                res += 1
                counter = 0
            counter |= m
        return res
